### PR TITLE
conformance ProvenCompactCiphertextList

### PR DESCRIPTION
**Pull request by mayeul-zama**

### CODE REVIEW

File: tfhe/src/shortint/ciphertext/zk.rs [hunk 1]
```diff
@@ -141,6 +146,74 @@ impl ProvenCompactCiphertextList {
     }
 }
 
+pub struct ProvenCompactCiphertextListConformanceParams {
+    pub sk_params: ClassicPBSParameters,
+    pub elements_per_block: usize,
```
Comment (tmontaigu) [inline tfhe/src/shortint/ciphertext/zk.rs:151]: Could this be called `elements_per_list/elements_per_vec/elements_per_sublist`, I think `block` may be too confusing with integer blocks

File: tfhe/src/high_level_api/compact_list.rs [hunk 2]
```diff
@@ -359,6 +363,61 @@ mod zk {
...
+        use crate::zk::CompactPkeCrs;
+
+        use super::*;
+        #[test]
+        fn conformanceaa() {
+            let mut rng = thread_rng();
+
+            let params: crate::shortint::ClassicPBSParameters =
+                crate::shortint::parameters::PARAM_MESSAGE_2_CARRY_2_KS_PBS_TUNIFORM_2M64;
+            let config = crate::ConfigBuilder::with_custom_parameters(params);
```
Comment (IceTDrinker) [inline tfhe/src/high_level_api/compact_list.rs:395]: need a test case with casting

File: tfhe/src/shortint/ciphertext/zk.rs [hunk 3]
```diff
@@ -141,6 +146,74 @@ impl ProvenCompactCiphertextList {
...
+                message_modulus: sk_params.message_modulus,
+                carry_modulus: sk_params.carry_modulus,
+                degree: Degree::new(sk_params.message_modulus.0 * sk_params.message_modulus.0 - 1),
+                noise_level: NoiseLevel::NOMINAL,
+                expansion_kind: *expansion_kind,
+            };
+
+            if !proof.is_valid() {
+                return false;
+            }
```
Comment (IceTDrinker) [inline tfhe/src/shortint/ciphertext/zk.rs:206]: this API name is confusing, so more "content_is_usable"

**Comment by IceTDrinker** [comment]
this has been merged as part of the release_pr_stack branch

**Pull request closed without merging**